test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs

```diff
@@ -100,10 +169,28 @@ public class UpdateSecretsManagerSubscriptionCommandTests
             ServiceAccountsAdjustment = 11
         };
 
+        var plan = StaticStore.SecretManagerPlans.Single(x => x.Type == organization.PlanType);
+        plan.HasAdditionalSeatsOption = true;
+        var newSeatCount = organization.SmSeats.GetValueOrDefault() + organizationUpdate.SeatAdjustment;
+
         var exception = await Assert.ThrowsAsync<BadRequestException>(
             () => sutProvider.Sut.UpdateSecretsManagerSubscription(organizationUpdate));
 
-        Assert.Contains("Cannot set max service account autoscaling below service account count.", exception.Message);
+        Assert.Contains("Cannot set max Service Accounts autoscaling below Service Accounts count", exception.Message);
+
+        var additionalSeats = newSeatCount - plan.BaseSeats;
+
+        await sutProvider.GetDependency<IPaymentService>().Received(1)
+            .AdjustSeatsAsync(organization, plan, additionalSeats);
+
+        await sutProvider.GetDependency<IReferenceEventService>().Received(1).RaiseEventAsync(Arg.Any<ReferenceEvent>());
+
+        await sutProvider.GetDependency<IPaymentService>().DidNotReceive()
+            .AdjustServiceAccountsAsync(Arg.Any<Organization>(), Arg.Any<Plan>(), Arg.Any<int>());
+
+        await sutProvider.GetDependency<IOrganizationService>().DidNotReceive().ReplaceAndUpdateCacheAsync(Arg.Any<Organization>());
+
+        await sutProvider.GetDependency<IMailService>().DidNotReceive().SendOrganizationMaxSeatLimitReachedEmailAsync(Arg.Any<Organization>(), Arg.Any<int>(), Arg.Any<IEnumerable<string>>());
```

Comment:
We discussed this on the call, but this is a good example of what we're trying to avoid. See how the paymentService received a request (to bill the customer) but `ReplaceAndUpdateCacheAsync` wasn't called (to update our database)? That would result in the customer getting billed for service accounts they can't use in the app. This is why we should only call Stripe (and events and database writes etc) at the end of all validation.

Reply:
done. please check it